Build.ps1

```diff
@@ -0,0 +1,129 @@
+Param(
+    # Archive containing the Ceph Windows binaries, will be fetched using scp.
+    # Can be a local path, a UNC path or a remote scp path.
+    [Parameter(Mandatory=$true)]
+    [string]$CephZipPath,
+
+    # The thumbprint of the X509 certificate used for signing the WNBD driver
+    # and the MSI installer
+    [string]$SignX509Thumbprint = $null
+)
+
+$ErrorActionPreference = "Stop"
+
+function SetVCVars($version="2019", $platform="x86_amd64") {
+    pushd "$ENV:ProgramFiles (x86)\Microsoft Visual Studio\$version\Community\VC\Auxiliary\Build"
+    try {
+        cmd /c "vcvarsall.bat $platform & set" |
+        foreach {
+          if ($_ -match "=") {
+            $v = $_.split("="); set-item -force -path "ENV:\$($v[0])"  -value "$($v[1])"
+          }
+        }
+    }
+    finally {
+        popd
+    }
+}
+
+function Expand7z($archive, $outputDir = ".") {
+    pushd .
+    try {
+        cd $outputDir
+        &7z.exe x -y $archive
+        if ($LastExitCode) { throw "7z.exe failed on archive: $archive"}
+    }
+    finally {
+        popd
+    }
+}
+
+function Sign($x509thumbprint, $path) {
+    & signtool.exe sign /sha1 $x509thumbprint /td SHA256 /v $path
+    if($LASTEXITCODE) { throw "signtool failed" }
+}
+
+function BuildWnbd() {
+
+    pushd $depsBuildDir
+    & git.exe clone https://github.com/cloudbase/wnbd

```

Comment:
nit: might be a good idea to use a wrapper such as this one: https://github.com/cloudbase/android-emulator-ci/blob/master/utils/windows/common.ps1#L65-L74

It spares us from checking LASTEXITCODE all the time, also printing the entire command in case of failures. More importantly, it prevents Powershell from treating stderr output as an exception. That usually happens when invoking scripts through WinRM.

Reply:
Since we dont face the winrm issue here and we rely only on the exit code, I'd just go with something like this for every non-PS command, but it's still a bit overkill here: https://github.com/cloudbase/cloudbase-init-installer/blob/master/BuildAutomation/BuildUtils.ps1#L300-L309